manubot/process/util.py

```diff
@@ -285,51 +307,50 @@ def get_citekeys_df(args, text):
             tag_df = pandas.read_csv(args.citation_tags_path, delim_whitespace=True)
         tag_df["manuscript_citekey"] = "tag:" + tag_df.tag
         tag_df = tag_df.rename(columns={"citation": "detagged_citekey"})
-        for detagged_citekey in tag_df.detagged_citekey:
-            is_valid_citekey(detagged_citekey, allow_raw=True)
-        citekeys_df = citekeys_df.merge(
-            tag_df[["manuscript_citekey", "detagged_citekey"]], how="left"
+        citekey_aliases = dict(
+            zip(tag_df["manuscript_citekey"], tag_df["detagged_citekey"])
         )
     else:
-        citekeys_df["detagged_citekey"] = None
+        citekey_aliases = {}
         logging.info(
             f"missing {args.citation_tags_path} file: no citation tags (citekey aliases) set"
         )
-    citekeys_df.detagged_citekey.fillna(
```

Comment:
Does removing this `fillna` change the final data frame?

Reply:
No the functionality has been moved to:

```python
    citekeys_df["detagged_citekey"] = citekeys_df.manuscript_citekey.map(
        lambda citekey: citekey_aliases.get(citekey, citekey)
    )
```